tests/ui/if_let_mutex.rs

```diff
@@ -9,7 +14,8 @@ fn do_stuff<T>(_: T) {}
 fn if_let() {
     let m = Mutex::new(1_u8);
     if let Err(locked) = m.lock() {
-        //~^ ERROR: calling `Mutex::lock` inside the scope of another `Mutex::lock` causes a d
+        //~[edition2021]^ if_let_mutex
+        // ERROR: calling `Mutex::lock` inside the scope of another `Mutex::lock` causes a d
```

Comment:
I think these ERROR: annotations are unused now that `//~[edition2021]` is used. Can they be deleted or are they kept intentionally?